### PR TITLE
Added support for elevation offset

### PR DESCRIPTION
**Pull request by bretwalker**

### CODE REVIEW

File: opencv_engine/engine.py [hunk 1]
```diff
@@ -135,13 +142,21 @@ def read_tiff(self, buffer, create_alpha=True):
...
+
                 red_channel = channels[0]
                 channels[0] = channels[2]
                 channels[2] = red_channel
 
+                if offset > 0:
+                    channels[2] = channels[2] + offset
+
+                    for i in range(len(channels)):
+                        channels[i] = channels[i] + 0.
```
Comment (jeremyeastwood) [inline opencv_engine/engine.py]: what is this doing?

File: opencv_engine/engine.py [hunk 2]
```diff
@@ -135,13 +142,21 @@ def read_tiff(self, buffer, create_alpha=True):
             channels = [gdal_img.GetRasterBand(i).ReadAsArray() for i in range(1, gdal_img.RasterCount + 1)]
 
             if len(channels) >= 3:  # opencv is bgr not rgb.
+
                 red_channel = channels[0]
                 channels[0] = channels[2]
                 channels[2] = red_channel
 
+                if offset > 0:
```
Comment (jeremyeastwood) [inline opencv_engine/engine.py]: we want to be able to potentially specify a negative offset here, so can either make it check `if offset` or just ignore the conditional

File: opencv_engine/engine.py (hunk 1 above)
Comment (bretwalker) [reply opencv_engine/engine.py]: At one point, the merge below was failing if one channels were float arrays and others were int arrays, but that's apparently not an issue anymore.

File: opencv_engine/engine.py [hunk 3]
```diff
@@ -135,13 +142,15 @@ def read_tiff(self, buffer, create_alpha=True):
             channels = [gdal_img.GetRasterBand(i).ReadAsArray() for i in range(1, gdal_img.RasterCount + 1)]
 
             if len(channels) >= 3:  # opencv is bgr not rgb.
+
                 red_channel = channels[0]
                 channels[0] = channels[2]
-                channels[2] = red_channel
+                channels[2] = red_channel + offset
```
Comment (cwsullivan) [inline opencv_engine/engine.py]: Hate to nit, but this bothers me. We're adding a unitless offset to a color. This is bound to confuse future generations. Is there any way we can clean this up?

Comment (jeremyeastwood) [reply opencv_engine/engine.py]: suggestion?

Comment (bretwalker) [reply opencv_engine/engine.py]: @cwsullivan Does this look okay?

```
            if len(channels) >= 3:  # opencv is bgr not rgb.

                red_channel = channels[0]
                channels[0] = channels[2]
                # If there's an offset, we are reading a DEM tiff, which stores its elevation data in channels[2]
                channels[2] = red_channel + offset
```

**Pull request merged**